db/dcrpg/pgblockchain.go

```diff
@@ -997,6 +997,36 @@ func (pgb *ChainDB) FillAddressTransactions(addrInfo *explorer.AddressInfo) erro
 			txn.Confirmations = 0
 		}
 		txn.FormattedTime = time.Unix(dbTx.BlockTime, 0).Format("2006-01-02 15:04:05")
+
+		// Get the funding or spending transaction matching index if there is a
+		// matching tx hash already present.  During the next database
+		// restructuring we may want to consider including matching tx index
+		// along with matching tx hash in the addresses table.
+		if txn.MatchedTx != `` {
+			if !txn.IsFunding {
+				// Lookup by the database row id
+				idx, err := RetrieveFundingOutpointIndxByVinID(pgb.db, dbTx.VinDbIds[txn.InOutID])
+
+				if err != nil {
+					log.Warnf("Matched Transaction Lookup failed for %s:%d: id: %d:  %v",
+						txn.TxID, txn.InOutID, txn.InOutID, err)
+				} else {
+					addrInfo.Transactions[i].MatchedTxIndex = idx
+				}
+
+			} else {
+				// Lookup by the matching tx hash and matching tx index
+				_, _, idx, _, err := RetrieveSpendingTxByTxOut(pgb.db, txn.TxID, txn.InOutID)
+
+				if err != nil {
+					log.Warnf("Matched Transaction Lookup failed for %s:%d: %v",
+						txn.TxID, txn.InOutID, err)
```

Comment:
The template triggers off of a value in matchedTx (the hash) and assuming we can get a matched tx index for every case that we have a matchedTx != '' then I think we are ok.  

Having said that as part of my testing this afternoon I confirmed that we do have some ambiguous results for this query and as such should add is_mainchain and is_valid to the selection criteria for this query.

```
SelectSpendingTxByPrevOut = `SELECT id, tx_hash, tx_index, tx_tree FROM vins
		WHERE prev_tx_hash=$1 AND prev_tx_index=$2 AND is_valid=TRUE AND is_mainchain=TRUE;`
```

Reply:
Ah, right I meant to bring that up.  Alternatively, `ORDER BY is_valid DESC, is_mainchain DESC, block_time DESC`.  This way, if there is no valid mainchain instance, you can at least link to the orphaned/invalidated transaction, which we are now tracking.  If there is a valid one, it will be first, and a valid mainchain one before valid side chain ones.  The block_time ordering is to go with the most recent one if there are no valid mainchain ones.